satpy/etc/readers/viirs_l2.yaml

```
@@ -104,6 +104,15 @@ datasets:
     file_type: cldprop_l2_viirs
     standard_name: cloud_top_height
 
+  Cloud_Optical_Thickness:
+    name: Cloud_Optical_Thickness
+    long_name: Cloud Optical Thickness two-channel retrieval using 2.2 um and either 0.65 um, 0.86 um or 1.24um (specified in Quality_Assurance) from best points: not failed in any way, not marked for clear sky restoral
```

If this value is in the file I don't think it should be needed here if the python code pulls it from the file and puts in in the DataArray's `.attrs`. If it is needed here, then it should be surrounded by double quotes:

```suggestion
    long_name: "Cloud Optical Thickness two-channel retrieval using 2.2 um and either 0.65 um, 0.86 um or 1.24um (specified in Quality_Assurance) from best points: not failed in any way, not marked for clear sky restoral"
```

---

True point, the long_name is covered in the netcdf. Shortened ncdump below:
```
short Cloud_Optical_Thickness(number_of_lines, number_of_pixels) ; 
    Cloud_Optical_Thickness:long_name = "Cloud Optical Thickness two-channel retrieval using 2.2 um and either 0.65 um, 0.86 um or 1.24um (specified in Quality_Assurance) from best points: not failed in any way, not marked for clear sky restoral" ; 
    Cloud_Optical_Thickness:_FillValue = -9999s ;
    Cloud_Optical_Thickness:valid_min = 0s ;
    Cloud_Optical_Thickness:valid_max = 15000s ;
    Cloud_Optical_Thickness:scale_factor = 0.01 ;
    Cloud_Optical_Thickness:add_offset = 0. ; 
    Cloud_Optical_Thickness:units = "none" ; 
```
My very first thought was to match it to `Cloud_Top_Height` and the variables in `aerdb_l2_viirs`, but I see the point in not having these information redundant. Can we just leave out the long_name tag or does it need to be specified, that this value should be taken from the `.attrs`?

Thanks for the review!
- Martin

---

You'd have to test it to be sure, but these lines in the python seem to do what we'd want:

https://github.com/pytroll/satpy/blob/5d2b1fd0da4ab0339abc2edb56b3c712da9f23c8/satpy/readers/viirs_l2.py#L102-L103

So if you leave `long_name` out of the YAML then the reader should pull it from the file's variable's attributes.